index.js

```diff
@@ -23,33 +24,230 @@ const state = {
+      price: 0.56
     },
     {
       id: "008-berry",
       name: "berry",
-      price: 0.35
+      price: 0.43
     },
     {
       id: "009-blueberry",
       name: "blueberry",
-      price: 0.35
+      price: 0.17
     },
     {
       id: "010-eggplant",
       name: "eggplant",
-      price: 0.35
+      price: 4.79
     }
   ],
   cart: []
-};
\ No newline at end of file
+};
+
+
+//Setting up the starting state for the filter
+state.selectedFilters = []
+const filters = {
+  vegtables: ['beetroot', 'carrot'],
+  fruits: ['apple', 'apricot', 'avocado', 'bananas', 'bell pepper', 'berry', 'blueberry', 'eggplant']
+}
+
+//Setting up the default list style(needs to be map to create a seperate copy)
+const defaultList = state.items.map(e => e)
+
+// Element selection 
+
+const storeList = document.querySelector('.store--item-list')
+const cartList = document.querySelector('.cart--item-list')
+const totalSpan = document.querySelector('.total-number')
+const filterButton = document.querySelector('.filter-button')
+const filterWindow = document.querySelector('.filter-window')
+const listStyle = document.querySelector('#list-style')
+
+
+// Display stores invetory
+const displayInvetory = () => {
+  //resets the store list (this is needed for the filter to update the store and show only the selected type)
+  storeList.innerHTML = ''
+
+  state.items.forEach(item => {
+    //checks if no filters a applied if so draw all items (this is the default behavior)
+    if (state.selectedFilters.length === 0) createStoreItem(item)
+    //if a filter is applied it will check which ones and only display the correspondig items
+    state.selectedFilters.forEach(filter => {
+      if (filter.includes(item.name)) createStoreItem(item)
+    })
+  })
+}
+
+//Creates a specific item for the store list
+const createStoreItem = (item) => {
+  const li = document.createElement('li')
+
+  const div = document.createElement('div')
+  div.className = 'store--item-icon'
+
+  const img = document.createElement('img')
+  img.src = `assets/icons/${item.id}.svg`
+  img.alt = item.name
+
+  const button = document.createElement('button')
+  button.innerText = 'Add to cart'
+  button.addEventListener('click', () => addToCart(item))
+
+  div.append(img)
+  li.append(div)
+  li.append(button)
+  storeList.append(li)
+}
+
+//Adds items to the cart
+const addToCart = (item) => {
+  //Checks if an item does not exists in the cart
+  if (!state.cart.includes(item)) {   //if so call addToCart to add to the list
+    item.quantity = 1
+    state.cart.push(item)
+  } else {                            //else increase quantity by 1
+    item.quantity++
+  }
+  //then call updateCart to display the newest state
+  updateCart()
+}
+
+// Create/Update the cart list
+const updateCart = () => {
+
+  //clears the list
+  cartList.innerHTML = ''
+
+  //keeps track of the carts total value
+  let cartTotal = 0
+
+  //for each item checks if quantity is > 0 
+  state.cart.forEach(item => {
```

Comment:
great use of forEach instead of creating a for loop and then targeting the current item with an index.